libs/shared/src/containers/TransactionList/index.tsx

```diff
@@ -53,77 +77,75 @@ export const TransactionList = (props: MyProps) => {
-        contentForComment = "OUT";
-      }
-      const monthTodisplay = tx.date.setLocale(locale).monthShort;
-      const yearToDisplay = tx.date.setLocale(locale).year;
-      const dayToDisplay = tx.date.setLocale(locale).day;
-      const timeToDisplay = tx.date.setLocale(locale).hour+":"+tx.date.setLocale(locale).minute;
-      return (
-        <Table.Row key={index}>
-          <Table.Cell width={2} textAlign='center'>
-            <div className={`${styles.dateInTable}`}>
-              <div className={`font-small write-vertical ${styles.yearInTable}`}>{yearToDisplay}</div>
-              <div className={"font-bold"}>{monthTodisplay}</div>
-              <div className={`font-big ${styles.dayInTable}`}>{dayToDisplay}</div>
-            </div>
-          </Table.Cell>
-          <Table.Cell width={1}><img src={imgForLine} /></Table.Cell>
-          <Table.Cell width={8}>
-            <div>{contentForComment}</div>
-            <span className={`font-small font-green font-bold`}>To</span> <span className={`font-grey-06`}>Test content</span>
-          </Table.Cell>
-          <Table.Cell width={3} textAlign='right'>
-            <div className={classForMoneyCell}>{changeCad} $</div>
-            <div className={`${styles.timeInTable}`}>{timeToDisplay}</div>
-          </Table.Cell>
-          <Table.Cell textAlign='right' width={3}><div className={`font-big`}>{balanceCad} $&nbsp;&nbsp;</div></Table.Cell>
-        </Table.Row>
-    )});
 
+  function onDateRangeChange(from: Date, until: Date) {
+    setFromDate(from);
+    setUntilDate(until);
+  }
+  const { locale } = useSelector(selectLocale);
+
+  const maxRowCount = 50;
+  const { rates } = props;
+  const account = useActiveAccount();
+  const transactions = account!.history;
+  const transactionLoading = account?.historyLoading;
+
+  let filteredTx = transactions.filter((tx) => tx.date.toMillis() >= fromDate.getTime() && tx.date.toMillis() <= untilDate.getTime())
+  filteredTx.reverse();
+  let [ txOutput, jsxFooter ] = buildPagination(filteredTx, maxRowCount, 0);
+
+  let txJsxRows = txOutput.map((tx, index) => {
+    const change = fiatChange(tx, rates);
+    const balance = tx.balance *  weBuyAt(rates, tx.date.toJSDate());
+    const changeCad = toHuman(change, true);
+    const balanceCad = toHuman(balance, true);
+
+    const imgForLine = changeCad < 0 ? iconCard : iconThecoin;
+    const classForMoneyCell = changeCad < 0 ? styles.moneyNegative : styles.moneyPositive;
+    const contentForComment = changeCad < 0 ? <FormattedMessage {...sent} /> : <FormattedMessage {...received} />;
+    const descForComment = changeCad < 0 ? <FormattedMessage {...to} /> : <FormattedMessage {...from} />;
+
+    const monthTodisplay = tx.date.setLocale(locale).monthShort;
+    const yearToDisplay = tx.date.setLocale(locale).year;
+    const dayToDisplay = tx.date.setLocale(locale).day;
+
+    const timeToDisplay = tx.date.setLocale(locale).toLocaleString(DateTime.TIME_SIMPLE);
```

Comment:
A lot of these are presentation components (ie, deciding how something is displayed, not what it means).  I would have pushed them all into the `TransactionLine`, basically so the TransactionLine can decide how it shows.  Partly separation-of-concerns, but also mostly because writing a story for the TxLine would be easier if it was in complete control how it displayed

Reply:
I mean pretty much everything from `imgForLine` down to `timeToDisplay`.  It's all the same data, just deciding what pixels to use

Reply:
I wanted to keep the component really short and stupid thought it would actually be easier for storybook